src/deploy.ts

```diff
@@ -200,18 +200,29 @@ export const deployPackage = async (
 };
 
 export const deployFromRepository = async (config: Config, plan: string) => {
-	//TODO fetch branch list via api and then show option rather than taking branch name
-	const branch = await input('Enter branch name');
-	//TODO check if repo url is valid
+	const api = API(config.token as string, config.baseURL);
+
+	let branch;
 	const url = args['addrepo'];
 
-	info(`Deploying from ${url as string}...\n`);
+	try {
+		const { branches } = await api.branchList(url as string);
```

Comment:
I guess url will always be of string type as it is mentioned in the args type, so you don't need to pass ```url as string``` everywhere.

Reply:
No type of url was <string | undefined> which was linter was not allowing to be a argument which should be a string type only

Reply:
So wrap the whole things like this ```if(typeof url  !== 'undefined')```
Do this only you think writing this way can reduce ```as``` from most of the places as code will look cleaner then.

Reply:
no did not solve this problem
i have tried